app/controllers/dashboard_controller.rb

```diff
@@ -14,6 +14,16 @@ def index
     end
   end
 
+  def show
+    if @widget.present? && @widget.user == User.current
+      render(:partial => @widget.template, :locals => @widget.data)
```

Comment:
it would be nice to log which widget we are rendering (not just the id) so we can troubleshoot more easily